utils/cp_utils.py

```diff
@@ -0,0 +1,71 @@
+"""
+This collection of functions runs CellProfiler and will rename the .sqlite outputed to any specified name if running an analysis pipeline.
+"""
+
+import os
+import pathlib
+
+
+def rename_sqlite_file(sqlite_dir_path: pathlib.Path, name: str):
+    """Rename the .sqlite file to be {name}.sqlite as to differentiate between the files.
+
+    Args:
+        sqlite_dir_path (pathlib.Path): path to CellProfiler_output directory
+        name (str): new name for the SQLite file
+    """
+    try:
+        # CellProfiler requires a name to be set in to pipeline, so regardless of plate or method, all sqlite files name are hardcoded
+        sqlite_file_path = pathlib.Path(f"{sqlite_dir_path}/NF1_data.sqlite")
+
+        new_file_name = str(sqlite_file_path).replace(
+            sqlite_file_path.name, f"{name}.sqlite"
+        )
+
+        # change the file name in the directory
+        pathlib.Path(sqlite_file_path).rename(pathlib.Path(new_file_name))
+        print(f"The file is renamed to {pathlib.Path(new_file_name).name}!")
+
+    except FileNotFoundError as e:
+        print(
+            f"The NF1_data.sqlite file is not found in directory. Either the pipeline wasn't ran properly or the file is already renamed.\n"
+            f"{e}"
+        )
+
+
+def run_cellprofiler(
+    path_to_pipeline: str,
+    path_to_output: str,
+    path_to_images: str,
+    sqlite_name: str = None,
+    analysis_run: bool = False,
```

Comment:
```python
from typing import Optional 

def foo( 
    arg1: str, 
    optional_arg1: Optional[None | str] = None, 
    optional_arg2: Optional[ None | str] = None)
)
```

These seem like optional parameters. try using pythons `Optional` from the `typing` module. Allow you to explicitly state that the parameter are optional parameters in your function.

Here's python's [documentation](https://docs.python.org/3/library/typing.html#typing.Optional)